drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/AbstractKogitoDMNService.java

```diff
@@ -107,60 +105,74 @@ protected ClientDMNType getDMNTypeFromMaps(final Map<String, ClientDMNType> dmnT
             }
         }
         /* Evaluating not primitive types defined into DMN file */
-        /* A sort of the items is mandatory to start the process. The reason is it needs to have dependent items,
-         * referred by typeRef field, BEFORE their referred items.  */
-        jsitItemDefinitions.sort(getItemDefinitionComparator());
+        final Map<String, JSITItemDefinition> indexed = indexDefinitionsByName(jsitItemDefinitions);
         for (int i = 0; i < jsitItemDefinitions.size(); i++) {
             final JSITItemDefinition jsitItemDefinition = Js.uncheckedCast(jsitItemDefinitions.get(i));
-            toReturn.put(jsitItemDefinition.getName(), getDMNType(jsitItemDefinition, nameSpace, toReturn));
+            toReturn.put(jsitItemDefinition.getName(), getDMNType(indexed, jsitItemDefinition, nameSpace, toReturn));
         }
+
+        toReturn.values().forEach((v) -> {
```

Comment:
@danielzhe Here, a `ConcurrentModificationException` can be thrown, because it iterates over  `toReturn` list which can be modified inside`populateItemFields(...)` method. I found it using the dmn file posted here by @dupliaka  https://issues.redhat.com/browse/KOGITO-1685.